engine/src/scheme.rs

```diff
@@ -214,10 +247,22 @@ impl<'s> Scheme {
     pub fn with_capacity(n: usize) -> Self {
         Scheme {
             fields: IndexMap::with_capacity_and_hasher(n, FnvBuildHasher::default()),
-            functions: Default::default(),
+            ..Default::default()
         }
     }
 
+    /// Retrieves the scheme configuration as immutable.
+    /// Usually used to read configuration values.
+    pub fn get_config(&self) -> &SchemeConfig {
+        &self.config
+    }
+
+    /// Retrieves the scheme configuration as mutable.
+    /// Usually used to write configuration values.
+    pub fn get_mut_config(&mut self) -> &mut SchemeConfig {
```

Comment:
The common notation is to add `mut` at the end, so it should be `get_config_mut`